core/metacling/src/TCling.cxx

```diff
@@ -5891,11 +5867,85 @@ Int_t TCling::AutoParse(const char *cls)
    return nHheadersParsed > 0 ? 1 : 0;
 }
 
+static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_name,
+      cling::Interpreter *fInterpreter) {
+   using namespace llvm::object;
+   using namespace llvm::sys::path;
+   using namespace llvm::sys::fs;
+
+   R__LOCKGUARD(gInterpreterMutex);
+
+   static bool sFirstRun = true;
+   // We don't want to do directory_iterator several times because it contains HDD access
+   static std::set<std::string> Libraries;
+
+   if (sFirstRun) {
+      // Store the information of path so that we don't have to iterate over the same path again and again.
+      std::unordered_set<std::string> AlreadyLookedPath;
+      const clang::Preprocessor &PP = fInterpreter->getCI()->getPreprocessor();
+      const HeaderSearchOptions &HSOpts = PP.getHeaderSearchInfo().getHeaderSearchOpts();
+      const auto ModulePaths(HSOpts.PrebuiltModulePaths);
+      cling::DynamicLibraryManager* dyLibManager = fInterpreter->getDynamicLibraryManager();
+
+      // Take path here eg. "/home/foo/module-release/lib/"
+      for (auto Path : ModulePaths) {
+         // Already searched?
+         if (AlreadyLookedPath.find(Path) != AlreadyLookedPath.end())
```

Comment:
Maybe `unordered_set::count` is more efficient.

Reply:
No, please just `insert` and check the return value `.second`!